<launch>

  <arg name="margin"                 default="0.1"/> <!-- [0 1] -->
  <arg name="confidence"             default="0.7"/>  <!-- (0.5 1) -->
  <arg name="collision_distance"     default="4.0"/>  <!-- m -->
  <arg name="heading_tol"            default="3.5"/>  <!-- agents don't turn before moving -->
  <arg name="minimum_update_rate"    default="0.5"/>   <!-- Hz -->
  <arg name="patrol_radius"          default="20.0"/> <!-- radius of patrol circle -->
  <arg name="source"                 default="7"/>
  <arg name="dest"                   default="4"/>

  <rosparam param="/task_agent_ids">[4,7]</rosparam>
  <rosparam param="/comm_agent_ids">[6]</rosparam>
  <param name="/desired_altitude" value="8.0"/>

  <!-- channel rate model params -->
  <param name="/N0" value="-70.0"/> <!-- -85.0  receiver noise (dBm) -->
  <param name="/n"  value="2.52"/>  <!--   2.52 decay exponent -->
  <param name="/L0" value="-52.0"/> <!-- -48.0  transmit power (dBm) -->
  <param name="/a"  value="0.2"/>   <!--   0.2  variance sigmoid param 1 -->
  <param name="/b"  value="0.6"/>   <!--   0.6  variance sigmoid param 2 -->

  <arg name="source_node"            default="$(arg source)"/>
  <arg name="start_x"                default="5.0"/>
  <arg name="start_y"                default="-10.0"/>
  <arg name="end_x"                  default="35.0"/>
  <param name="/source_node"      value="7"/>
  <param name="/start_x"          value="$(arg start_x)"/>
  <param name="/start_y"          value="$(arg start_y)"/>
  <param name="/end_x"            value="$(arg end_x)"/>

  <node pkg="network_planner"
        type="network_planner_routing_node"
        name="network_planner_routing_node"
        output="screen">
    <param name="sample_count"        value="0"/>
    <param name="sample_variance"     value="0.2"/>
    <param name="source"              value="$(arg source)"/>
    <param name="dest"                value="$(arg dest)"/>
    <param name="margin"              value="$(arg margin)"/>
    <param name="confidence"          value="$(arg confidence)"/>
    <param name="collision_distance"  value="0.2"/>
    <param name="minimum_update_rate" value="$(arg minimum_update_rate)"/>
    <param name="pose_topic_prefix"   value=""/>
    <param name="pose_topic_suffix"   value="/pose"/>
    <param name="nav_nodelet"         value="px4_waypoint_navigation_nodelet"/>
    <param name="world_frame"         value="world"/>
  </node>

</launch>
